### PR TITLE
Feature/fix timeout

### PR DESCRIPTION
**Pull request by smihica**
Supported %read,%write timeout on windows.
UNIX is not supported the feature yet.
But this commit will not break UNIX.

### CODE REVIEW

**Comment by fukamachi** [comment]
Thanks for the patch.
I have some feedbacks.

I'm not sure returning `NIL` in `%read` and `%write` when the operation timeout is a good idea.
They are expected to return a `word` in most cases, so people tend to write a program which expects those functions to always return a `word`. If timeout happened in such a program, `TYPE-ERROR` would be raised.
It's not a good idea from the point of view of optimization because we can't determine the return-type of the functions.

There're two options.

1. Raise a `TIMEOUT-ERROR` like trivial-timeout.
2. Return multiple values with `0` or `#()`, and `T` or `NIL` as a timeout flag.

Besides that, it would be better that the timeout feature can be used with Gray Streams interface.
Since `read-sequence`, `write-sequence` or other CL's I/O operations doesn't take `:timeout`, I guess introducing a special variable like `*default-timeout*` and binding it before I/O is the only way to do it.

By considering the implementation, plan 1 (`with-timeout` and `timeout-error`) looks good to me.

**Comment by fukamachi** [comment]
NOTE: The UNIX version of this might use `select(2)`.

**Comment by smihica** [comment]
Nice idea. I'll implement the 1st choice.
Thank you.

File: src/main.lisp [hunk 1]
```diff
@@ -162,3 +160,7 @@ If timeout is non-nil then the function will return nil after that many seconds
      (unwind-protect
 	  (progn ,@body)
        (close-serial ,serial))))
+
+(defmacro with-timeout ((ms) &body body)
+  `(let ((*default-timeout* ,ms))
```
Comment (fukamachi) [inline src/main.lisp]: Probably `*default-timeout-ms*` is correct?

Comment (smihica) [inline src/main.lisp]: oh no..

**Comment by smihica** [comment]
I've done. thank you.

**Pull request merged**